src/server/job_func.c

```diff
@@ -708,27 +708,61 @@ del_job_related_file(job *pjob, char *fsuffix)
 }
 
 #ifdef PBS_MOM
+/**
+ * @brief	Rename the job's <taskdir>.TK to <taskdir>.TK.RM
+ *
+ * @param[in]	pjob - the job being purged
+ *
+ * @return	char *
+ * @retval	!NULL  - new taskdir path name, malloced which must be freed.
+ * @retval	NULL - rename failed, error encountered
+ */
+static char *
+rename_taskdir(job *pjob)
+{
+	char namebuf[MAXPATHLEN + 1] = {'\0'};
+	char renamebuf[MAXPATHLEN + 1] = {'\0'};
+	char *fprefix;	
+
+	if (pjob == NULL)
+		return (NULL);
+
+	if (*pjob->ji_qs.ji_fileprefix != '\0')
+		fprefix = pjob->ji_qs.ji_fileprefix;
+	else
+		fprefix = pjob->ji_qs.ji_jobid;
+
+	snprintf(namebuf, sizeof(namebuf) - 1, "%s%s%s", path_jobs, fprefix, JOB_TASKDIR_SUFFIX);
```

Comment:
Ok, I'll switch to using pbs_strcpy()...

Reply:
Can this all be wrapped up into one pbs_asprintf()?

Reply:
@bhroam : Hiren suggested I use pbs_strcpy() instead as it is more efficient. I'm not sure how that compares to pbs_asprintf()...

Reply:
Unless this is deep in a nested loop, I would say err on the side of readability over speed.  That is just me though.

Reply:
Sorry I missed last conversion on this... @bayucan we can use pbs_asprintf also as it is almost identical to malloc + strcpy/cat (except it will do sprinf() like stuff)

@bhroam well this is not deep in a nested loop, but this will run per job end

Reply:
@hirenvadalia : I've switched to pbs_asprintf() and re-tested. Please re-review @minghui-liu . Thanks.